File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/SDDAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface SDDAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // SDD Params
    default SDDAttributes setIban(String iban) {
        requestBuilder.addElement("iban", iban);
        return this;
    }

    default SDDAttributes setBic(String bic) {
        requestBuilder.addElement("bic", bic);
        return this;
    }

    default RequestBuilder buildSDDParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/SDDAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface SDDAttributes {

    // SDD Params
    fun setIban(iban: String): SDDAttributes {
        requestBuilder.addElement("iban", iban)
        return this
    }

    fun setBic(bic: String): SDDAttributes {
        requestBuilder.addElement("bic", bic)
        return this
    }

    fun buildSDDParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}
